orangecontrib/text/widgets/owpubmed.py

```diff
@@ -0,0 +1,463 @@
+
+    def enable_controls(self):
+        # Enable/disable controls accordingly.
+        for control in self.pubmed_controls:
+            control.setEnabled(self.email_is_valid)
+        if self.pubmed_api is None or self.pubmed_api.search_record_count == 0:
+            self.retrieve_records_button.setEnabled(False)
+        if not self.email_is_valid:
+            self.email_combo.setFocus()
+
+    def run_search(self):
+        self.warning(0)
+        self.run_search_button.setEnabled(False)
+        self.retrieve_records_button.setEnabled(False)
+
+        # Add the email to history.
+        email = self.email_combo.currentText()
+        if email not in self.recent_emails:
+            self.recent_emails.insert(0, email)
+
+        # Check if the PubMed object is present.
+        if self.pubmed_api is None:
+            self.pubmed_api = Pubmed(
+                    email=email,
+                    progress_callback=self.api_progress_callback,
+                    error_callback=self.api_error_callback,
+            )
+
+        if self.search_tabs.currentIndex() == 0:
+            # Get query parameters.
+            terms = self.keyword_combo.currentText().split()
+            authors = self.author_input.text().split()
+
+            # If no keywords, alert that the query is too vague.
+            if not terms:
+                self.warning(0, 'Please specify the keywords for this query.')
+                self.run_search_button.setEnabled(True)
+                self.retrieve_records_button.setEnabled(True)
+                return
+
+            # Check date formatting.
+            pdate_from_flag = self.pub_date_from and not eval_date(
+                    self.pub_date_from
+            )
+            pdate_to_flag = self.pub_date_to and not eval_date(
+                    self.pub_date_to
+            )
+            if pdate_from_flag or pdate_to_flag:
+                self.warning(0, 'Please specify dates with digits in '
+                                'YYYY/MM/DD format.')
+                self.run_search_button.setEnabled(True)
+                self.retrieve_records_button.setEnabled(True)
+                return
+
+            error = self.pubmed_api._search_for_records(terms,
+                                                        authors,
+                                                        self.pub_date_from,
+                                                        self.pub_date_to)
+            if error is not None:
+                self.warning(0, str(error))
+                return
+
+            if self.keyword_combo.currentText() not in self.recent_keywords:
+                self.recent_keywords.insert(
+                        0,
+                        self.keyword_combo.currentText()
+                )
+        else:
+            query = self.advanced_query_input.toPlainText()
+            if not query:
+                self.warning(0, 'Please specify the keywords for this query.')
```

Comment:
Are keyword really required? Can't we search for example just by author? If so, change to something like `Please specify the query.`